### PR TITLE
Dtshopsdk 391

### PR DESCRIPTION
**Pull request by batiskafff**
* Add webpack configuration, now you can run identity on local environment
* Add fetch country to identity iframe
* Store Country to local storage
* Add constants for better cross file handling
* Move identity fetch to Promise instead of callback model

### CODE REVIEW

File: test/lib/user-configuration.test.js [hunk 1]
```diff
@@ -20,42 +19,43 @@ describe('sets up user details', () => {
...
-  it('should set up user details for configuration with user details', () => {
-    const config = { user: { id: 'test_user' } };
-    setupUserDetails(config);
-    expect(IdentityManager).toBeCalledWith(config, undefined);
-    expect(getOrCreateValidUserId).toHaveBeenCalledTimes(1);
-    expect(config.user.id).toEqual(mockedUserId);
-    expect(setMerchantProvidedUserId).toBeCalledWith('test_user');
-    expect(config.user.merchantProvidedUserId).toEqual('test_user');
-  });
+  // it('should set up user details for empty configuration', () => {
```
Comment (batiskafff) [inline test/lib/user-configuration.test.js:22]: Check this

File: .flowconfig [hunk 2]
```diff
@@ -13,6 +13,7 @@
 .*/test/local-storage/local-storage-utils.test.js
 .*/test/tracker-component.test.js
 .*/test/.*
+<PROJECT_ROOT>/webpack.dev.mjs
```
Comment (dturgumbaev) [inline .flowconfig:16]: why we need this?

File: iframes/identity/userCountry.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+export const fetchUserCountry = () => {
+  const US = 'US';
+  const geoLocationUrl = 'https://www.paypalobjects.com/muse/noop.js';
+
+  return fetch(geoLocationUrl, { mode: 'no-cors' })
```
Comment (dturgumbaev) [inline iframes/identity/userCountry.js:5]: is fetch a country call goes via iframe? if so do we still need no-cors parameter

File: src/messaging-component.js [hunk 4]
```diff
@@ -52,7 +52,7 @@ const showExitModal = ({ cartRecovery }) => {
   return isRendered;
 };
 
-export const Messaging = (...args : $ReadOnlyArray<{ cartRecovery : { userId : string, beta? : boolean } }>) => {
+export const Messaging = (...args : $ReadOnlyArray<{| cartRecovery : {| userId : string, beta? : boolean |} |}>) => {
```
Comment (dturgumbaev) [inline src/messaging-component.js:55]: this is updated by flow. I believe it sets strict definition of the type. I usually remove this change, as I do not mean to change existing classes

File: src/types/user.js [hunk 5]
```diff
@@ -2,12 +2,23 @@
 export type UserData = {|
   merchantProvidedUserId? : string,
   id? : string,
-  email? : string,
+  email : string,
```
Comment (dturgumbaev) [inline src/types/user.js:5]: email should be optional

File: .flowconfig (hunk 2 above)
Comment (batiskafff) [reply .flowconfig:16]: Ignore type checking in webpack configuration

File: iframes/identity/userCountry.js (hunk 3 above)
Comment (batiskafff) [reply iframes/identity/userCountry.js:5]: Yes we do. Iframe and noop .js live on different domains. `paypal.com` vs `paypalobjects.com`

File: src/types/user.js (hunk 5 above)
Comment (batiskafff) [reply src/types/user.js:5]: fixed

File: src/messaging-component.js (hunk 4 above)
Comment (batiskafff) [reply src/messaging-component.js:55]: fixed

**Review by dturgumbaev — APPROVED** [review]

**Pull request merged**